### ISSUE TITLE
Sometimes it reload only 1 ammo

### ISSUE THREAD

**Issue by Chryst** [ISeeDEDPpl/Questor#134]

**Comment by ISeeDEDPpl** [comment]
which branch?

**Comment by Chryst** [comment]
experimental

**Comment by grdk** [comment]
I'm using FixReloadAllIssue right now.

As a semi-kinda-yeah extension to this bug... if it runs out of ammo during a mission, then goes back to the station to rearm, AND the mission objective happens to be completed already, it does this...

00:07:52 [Salvage] Locking [Gist Malakim Wreck][7k][ID:9000000000001791975][7k away]
00:07:52 [Combat] Out of Ammo!
00:07:54 [TravelerDestination.StationDestination] Destination set to [Destination Station]
00:07:54 [TravelerDestination.StationDestination] Warp to and dock at [Destination Station][1.87 AU away]
00:08:36 [Defense] Shields: [100%] Cap: [28%] Shield Booster: [1] deactivated [2] sec reactivation delay
00:08:40 [TravelerDestination.StationDestination] Dock at [Destination Station] which is [-2k away]
00:08:48 [TravelerDestination.StationDestination] Arrived in station
00:08:49 [CombatMissionsBehavior] UnloadLoot: Begin
00:09:04 [Cache.OpenCorpLootHangar] ExpandCorpHangar executed
00:09:06 [Cache.OpenCorpLootHangar] ExpandCorpHangar executed
00:09:07 [Cache.OpenCorpLootHangar] ExpandCorpHangar executed
00:09:08 [Cache.OpenCorpLootHangar] SelectHangarID: 1
00:09:09 [Cache.OpenAmmoHangar] OpenCorpLootHangar: Open Secondary Window
00:09:12 [Cache.OpenAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:09:13 [Cache.OpenAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:09:13 [UnloadLoot] Moving Ammo to AmmoHangar [('InventorySecondary_StationCorpHangar', '1486965393_1')]
00:09:13 [UnloadLoot] Waiting for items to move
00:09:17 [Cache.StackCorpAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:09:17 [Cache.StackCorpAmmoHangar] Stacking Corporate Ammo Hangar: waiting [3sec]
00:09:17 [UnloadLoot.StackHangars] Closing Corp Window: Resupply
00:09:20 [Cache.OpenCorpLootHangar] SelectHangarID: 2
00:09:21 [Cache.OpenCorpLootHangar] OpenCorpLootHangar: Open Secondary Window
00:09:24 [Cache.OpenCorpLootHangar] OpenCorpLootHangar: Window is ready
00:09:24 [UnloadLoot] Moving [75] of [709] items into the loot hangar
00:09:24 [UnloadLoot] Loot was worth an estimated [25,962,094] isk in buy-orders
00:09:29 [UnloadLoot] Stacking items in Ammo Hangar: resuming in [ 3 sec ]
00:09:32 [Cache.StackCorpLoothangar] OpenCorpLootHangar: Window is ready
00:09:32 [Cache.StackLootHangar] Stacking Corporate Loot Hangar: waiting [3sec]
00:09:32 [UnloadLoot.StackHangars] Closing Corp Window: Drop
00:09:33 [UnloadLoot] Done
00:09:33 [CombatMissionsBehavior.UnloadLoot] We are out of ammo
00:09:34 [Statistics] It is unlikely a mission has been run yet this session... No Mission log needs to be written.
00:09:35 [CombatMissionsBehavior] Random start delay of [39] seconds
00:10:15 [Questor] EVE instance: totalMegaBytesOfMemoryUsed - 1466 MB
00:10:16 [AgentInteraction] Start conversation [Start Mission]
00:10:16 [AgentInteraction] Waiting for conversation
00:10:18 [AgentInteraction] Replying to agent
00:10:26 [AgentInteraction] Saying [View Mission]
00:10:35 [AgentInteraction] Saying [Complete Mission]
00:10:35 [AgentInteraction] Closing conversation
00:10:36 [Questor] Wallet Balance Has Not Changed in [ 2 ] minutes.
00:10:40 [AgentInteraction] Attempting to close Agent Window
00:10:42 [AgentInteraction] Done
00:10:43 [Arm] Begin
00:10:45 [Arm] Opening Ship Hangar: waiting [4sec]
00:10:49 [Arm] Activating combat ship
00:10:51 [Cache.OpenCorpLootHangar] SelectHangarID: 1
00:10:51 [Arm] Opening ammo hangar
00:10:52 [Cache.OpenAmmoHangar] OpenCorpLootHangar: Open Secondary Window
00:10:52 [Arm] Opening ammo hangar
00:10:53 [Arm] Opening ammo hangar
00:10:54 [Arm] Opening ammo hangar
00:10:55 [Arm] Opening ammo hangar
00:10:56 [Cache.OpenAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:10:56 [Arm] Moving Drones
00:10:57 [Arm] Opening Drone Bay: waiting [3sec]
00:11:00 [Cache.OpenAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:11:00 [Arm] No Drones with armor damage found in your DroneBay, you must be doing something right.
00:11:00 [Arm] neededDrones: 2
00:11:00 [Arm] Move [ 2 ] Drones into drone bay
00:11:01 [Cache.OpenAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:11:01 [Arm] No Drones with armor damage found in your DroneBay, you must be doing something right.
00:11:01 [Arm] neededDrones: 0
00:11:02 [Cache.OpenAmmoHangar] OpenCorpAmmoHangar: Window is ready
00:11:02 [Arm] Moving [1] units of Ammo [Nova Javelin Torpedo] from [('InventorySecondary_StationCorpHangar', '1486965393_1')] to CargoHold
00:11:02 [Arm] Waiting for items
00:11:08 [Arm] Done
00:11:09 [Arm] Done
00:11:10 [Arm] Done
00:11:11 [Arm] Done

**Comment by ISeeDEDPpl** [comment]
ok, found this bug... when we changed the agentinteractionpurpose 'on the fly' in agentinteraction we were not communicating that change back up through the stack. The cmission was complete but the combat mission behavior did not know that. It was still executing the "CombatMissionsBehaviorState.Start:" state which makes sure were have a mission a (the agent interaction part) and then arms the ship and runs out in space to do it. 

I changed "CombatMissionsBehaviorState.Start:" to detect the change in agentinteractionpurpose and fix the faulty behavior.  

I'll commit the fix after some testing.
